client/verta/verta/_deployment/endpoint.py

```diff
@@ -171,6 +170,24 @@ def update(self, run, strategy, wait=False, resources=None, autoscaling=None, en
 
         return self.get_status()
 
+    def _create_build(self, model_version_id=None, run_id=None):
```

Comment:
I think this should _also_ just take `model_reference` as a single parameter. Otherwise, we're basically duplicating the same general check on lines 140, 174, and 183. Instead, we can just have one check at line 183.

Reply:
Yeah but how can we tell if we should pass it to "model_version_id" or "run_id"?

Reply:
If it takes `model_reference`, check its type like you're doing right now on line 140.

Reply:
Oh I see. Will fix.

Reply:
Basically, my suggestion is to consolidate these three checks that are being done right now into one by passing the `model_reference` object into this helper function, and doing all the checking logic inside.

Reply:
Fixed. Not quite sure how to fix the check in update_from_config though.